src/senaite/core/content/supplier.py

```diff
@@ -0,0 +1,257 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of SENAITE.CORE.
+#
+# SENAITE.CORE is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2024 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+from AccessControl import ClassSecurityInfo
+from bika.lims import senaiteMessageFactory as _
+from bika.lims.interfaces import IDeactivable
+from bika.lims.validators import country_dic
+from bika.lims.validators import letter_dic
+from bika.lims.validators import _toIntList
+from bika.lims.validators import _sumLists
+from senaite.core.i18n import translate
+from senaite.core.catalog import SETUP_CATALOG
+from senaite.core.content.organization import IOrganizationSchema
+from senaite.core.content.organization import Organization
+from senaite.core.interfaces import ISupplier
+from plone.supermodel import model
+from Products.CMFCore import permissions
+from Products.CMFPlone.utils import safe_unicode
+from zope import schema
+from zope.interface import Invalid
+from zope.interface import implementer
+from zope.interface import invariant
+
+
+class ISupplierSchema(IOrganizationSchema):
+    """Schema interface
+    """
+
+    lab_account_number = schema.TextLine(
+        title=_(
+            u"title_supplier_lab_contact_number",
+            default=u"Lab Account Number",
+        ),
+        required=False,
+    )
+
+    remarks = schema.Text(
+        title=_(
+            u"title_supplier_remarks",
+            default=u"Remarks",
+        ),
+        required=False,
+    )
+
+    website = schema.TextLine(
+        title=_(
+            u"title_supplier_website",
+            default=u"Website",
+        ),
+        required=False,
+    )
+
+    model.fieldset(
+        "bank_details",
+        label=_(
+            u"Accounting",
+            default=u"Bank Details"
+        ),
+        fields=[
+            "nib",
+            "iban",
+            "swift_code",
+        ]
+    )
+
+    nib = schema.TextLine(
+        title=_(
+            u"title_supplier_nib",
+            default=u"National Identification Bank Account Number",
+        ),
+        required=False,
+    )
+
+    iban = schema.TextLine(
+        title=_(
+            u"title_supplier_iban",
+            default=u"International Bank Account Number",
+        ),
```

Comment:
Better keep "IBAN" as title and use "International Bank Account Number" as description.